### ISSUE TITLE
[Feature Request] Modpack Friendly

### ISSUE THREAD

**Issue by itsdinkd** [gamma-delta/HexMod#131]
As a modpack dev, I want to control what can and cannot be casted. Seeing the example videos from the mod page, there's quite a few castings that i'd like to blacklist/ban, but as it stands, there is no way to do this. Please add in a config entry to disable certain combos or something similar to where the modpack dev or a server operator can control what can and cannot be casted.

**Comment by itsdinkd** [comment]
Now actually seeing the server config file, there is a way.

However, can you please provide a list of all the circles and what they do so I can pick which ones to disable? Might

**Comment by gamma-delta** [comment]
What do you mean by "a list of all the circles"?

The action names can be found in [`RegisterPatterns.java`](https://github.com/gamma-delta/HexMod/blob/main/Common/src/main/java/at/petrak/hexcasting/common/casting/RegisterPatterns.java)

**Comment by itsdinkd** [comment]
> What do you mean by "a list of all the circles"?
> 
> The action names can be found in [`RegisterPatterns.java`](https://github.com/gamma-delta/HexMod/blob/main/Common/src/main/java/at/petrak/hexcasting/common/casting/RegisterPatterns.java)

There are two blacklist arrays in the server config, Circle Actions and Actions

**Comment by TacocaTYT** [comment]
> > What do you mean by "a list of all the circles"?
> 
> > 
> 
> > The action names can be found in [`RegisterPatterns.java`](https://github.com/gamma-delta/HexMod/blob/main/Common/src/main/java/at/petrak/hexcasting/common/casting/RegisterPatterns.java)
> 
> 
> 
> from that file, would this be a circle action? `qqaawawaeqqdd`

i see your confusion. there are two blacklists, a blacklist for spells that can be cast by the player (which probably also stops circles from casting them), and a list for stopping only spell circles from using that spell. this was because (as brought up in another issue) a player could set up a circle to make it always one type of weather or always one time of day and then just afk, which is annoying to other players on the server. but at the same time, we want active players to maybe be allowed to change the weather or time using trinkets or manual spells. 

TL;DR circle spells is referring to what spells can spell circles not cast but players can

**Comment by TacocaTYT** [comment]
also the book contains literally every action, or alternatively they are all [listed here](https://gamma-delta.github.io/HexMod/)

**Comment by itsdinkd** [comment]
> also the book contains literally every action, or alternatively they are all [listed here](https://gamma-delta.github.io/HexMod/)

Yeaaah I checked the book but unless you spend tons of time reading through each entry (this is the technical part of the mod), you will not understand it. I am trying to get this released with my modpack's update, which I do plan on going through every portion of the book, but before doing that, i'd like to just get what I feel is OP out of the way first so I can release and not delay.

**Comment by TacocaTYT** [comment]
what do you feel is op? a lot of this mod is balanced behind the scenes but if you tell us what you want (come join the discord for faster and better help) we can probably help you achieve what you want

**Comment by itsdinkd** [comment]
> > > What do you mean by "a list of all the circles"?
> > 
> > 
> > > 
> > 
> > 
> > > The action names can be found in [`RegisterPatterns.java`](https://github.com/gamma-delta/HexMod/blob/main/Common/src/main/java/at/petrak/hexcasting/common/casting/RegisterPatterns.java)
> > 
> > 
> > from that file, would this be a circle action? `qqaawawaeqqdd`
> 
> i see your confusion. there are two blacklists, a blacklist for spells that can be cast by the player (which probably also stops circles from casting them), and a list for stopping only spell circles from using that spell. this was because (as brought up in another issue) a player could set up a circle to make it always one type of weather or always one time of day and then just afk, which is annoying to other players on the server. but at the same time, we want active players to maybe be allowed to change the weather or time using trinkets or manual spells.
> 
> TL;DR circle spells is referring to what spells can spell circles not cast but players can

Okay so what I would need to do is blacklist spells, which are just "actions" correct? What part of the RegisterPatterns would I enter in the array? the `modLoc` value? 

i went to the link you sent of that wiki, however, I am not seeing action names? what are the values I use for the blacklist?

**Comment by itsdinkd** [comment]
> what do you feel is op? a lot of this mod is balanced behind the scenes but if you tell us what you want (come join the discord for faster and better help) we can probably help you achieve what you want

okay on my way

**Comment by gamma-delta** [comment]
Guess I'll close this?
